crates/bevy_sprite/src/frustum_culling.rs

```diff
@@ -0,0 +1,108 @@
+use bevy_asset::{Assets, Handle};
+use bevy_ecs::prelude::{Commands, Entity, Query, Res, With};
+use bevy_math::Vec2;
+use bevy_render::{camera::Camera, draw::Viewable};
+use bevy_transform::components::Transform;
+use bevy_window::Windows;
+
+use crate::{Sprite, TextureAtlas, TextureAtlasSprite};
+
+struct Rect {
+    position: Vec2,
+    size: Vec2,
+}
+
+impl Rect {
+    #[inline]
+    pub fn is_intersecting(&self, other: Rect) -> bool {
+        self.position.distance(other.position) < (self.get_radius() + other.get_radius())
+    }
+
+    #[inline]
+    pub fn get_radius(&self) -> f32 {
+        let half_size = self.size / Vec2::splat(2.0);
+        (half_size.x.powf(2.0) + half_size.y.powf(2.0)).sqrt()
+    }
+}
+
+pub fn frustum_culling_sprites(
+    mut commands: Commands,
+    windows: Res<Windows>,
+    cameras: Query<&Transform, With<Camera>>,
+    visible: Query<&Viewable, With<Sprite>>,
+    sprites: Query<(Entity, &Transform, &Sprite)>,
+) {
+    let window = windows.get_primary().unwrap();
```

Comment:
Just making sure/highlighting if we're okay with unwrapping in Bevy-provided systems (as opposed to logging and exiting cleanly)?

Reply:
dealt with unwrap to be on the safe side.